decompiler/structures/ast/condition_symbol.py

```diff
@@ -81,7 +81,10 @@ def get_z3_condition_map(self) -> Dict[LogicCondition, PseudoLogicCondition]:
 
     def add_condition(self, condition: Condition) -> ConditionSymbol:
         """Adds a new condition to the condition map and returns the corresponding condition_symbol"""
+        ssa_names_of_variables = {v.ssa_name for v in condition.requirements}
         for value in self._condition_map.values():
+            if ssa_names_of_variables != {v.ssa_name for v in value.condition.requirements}:
+                continue
```

Comment:
This check is pretty insane and pretty hard on runtime as well. Care to elaborate what is happening here?

Reply:
I changed it and use the pseudo Condition.
Furthermore, I updated it to return the symbol, because ConditionSymbol only should contain something line `x1` as symbol and not `!x1`. At least we assumed it at some point I guess.